<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <author>Massimo Callegari</author>
 <comment>
  Q Light Controller Plus
  vcspeeddialproperties.ui

  Copyright (c) 2015 Massimo Callegari

  Licensed under the Apache License, Version 2.0 (the &quot;License&quot;);
  you may not use this file except in compliance with the License.
  You may obtain a copy of the License at

      http://www.apache.org/licenses/LICENSE-2.0.txt

  Unless required by applicable law or agreed to in writing, software
  distributed under the License is distributed on an &quot;AS IS&quot; BASIS,
  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  See the License for the specific language governing permissions and
  limitations under the License.
 </comment>
 <class>VCSpeedDialProperties</class>
 <widget class="QDialog" name="VCSpeedDialProperties">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>551</width>
    <height>612</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Speed Dial Properties</string>
  </property>
  <property name="windowIcon">
   <iconset resource="../qlcui.qrc">
    <normaloff>:/speed.png</normaloff>:/speed.png</iconset>
  </property>
  <layout class="QGridLayout" name="gridLayout_2">
   <item row="0" column="0">
    <widget class="QLabel" name="label">
     <property name="text">
      <string>Speed Dial Name</string>
     </property>
    </widget>
   </item>
   <item row="0" column="1">
    <widget class="QLineEdit" name="m_nameEdit">
     <property name="toolTip">
      <string>Title of the dial</string>
     </property>
    </widget>
   </item>
   <item row="1" column="0" colspan="2">
    <widget class="QTabWidget" name="tabWidget">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="Functions">
      <attribute name="title">
       <string>Functions</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout">
       <item row="0" column="2">
        <widget class="QToolButton" name="m_addButton">
         <property name="toolTip">
          <string>Add functions to be controlled</string>
         </property>
         <property name="icon">
          <iconset resource="../qlcui.qrc">
           <normaloff>:/edit_add.png</normaloff>:/edit_add.png</iconset>
         </property>
         <property name="iconSize">
          <size>
           <width>32</width>
           <height>32</height>
          </size>
         </property>
        </widget>
       </item>
       <item row="1" column="2">
        <widget class="QToolButton" name="m_removeButton">
         <property name="toolTip">
          <string>Remove selected functions</string>
         </property>
         <property name="icon">
          <iconset resource="../qlcui.qrc">
           <normaloff>:/edit_remove.png</normaloff>:/edit_remove.png</iconset>
         </property>
         <property name="iconSize">
          <size>
           <width>32</width>
           <height>32</height>
          </size>
         </property>
        </widget>
       </item>
       <item row="2" column="2">
        <widget class="QToolButton" name="m_copyFactorsButton">
         <property name="toolTip">
          <string>Copy the factors of the selected function</string>
         </property>
         <property name="icon">
          <iconset resource="../qlcui.qrc">
           <normaloff>:/editcopy.png</normaloff>:/editcopy.png</iconset>
         </property>
         <property name="iconSize">
          <size>
           <width>32</width>
           <height>32</height>
          </size>
         </property>
        </widget>
       </item>
       <item row="3" column="2">
        <widget class="QToolButton" name="m_pasteFactorsButton">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="toolTip">
          <string>Paste copied factors to all selected functions</string>
         </property>
         <property name="icon">
          <iconset resource="../qlcui.qrc">
           <normaloff>:/editpaste.png</normaloff>:/editpaste.png</iconset>
         </property>
         <property name="iconSize">
          <size>
           <width>32</width>
           <height>32</height>
          </size>
         </property>
        </widget>
       </item>
       <item row="4" column="2">
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>18</width>
           <height>198</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="0" column="0" rowspan="5" colspan="2">
        <widget class="QTreeWidget" name="m_tree">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
           <horstretch>0</horstretch>
           <verstretch>1</verstretch>
          </sizepolicy>
         </property>
         <property name="alternatingRowColors">
          <bool>true</bool>
         </property>
         <property name="selectionMode">
          <enum>QAbstractItemView::ExtendedSelection</enum>
         </property>
         <property name="rootIsDecorated">
          <bool>false</bool>
         </property>
         <property name="itemsExpandable">
          <bool>false</bool>
         </property>
         <property name="sortingEnabled">
          <bool>true</bool>
         </property>
         <property name="animated">
          <bool>true</bool>
         </property>
         <property name="allColumnsShowFocus">
          <bool>true</bool>
         </property>
         <property name="expandsOnDoubleClick">
          <bool>false</bool>
         </property>
         <column>
          <property name="text">
           <string>Function</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Fade In factor</string>
          </property>
          <property name="toolTip">
           <string>Multiplier applied before time is sent as Fade In Time to the function.</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Fade Out factor</string>
          </property>
          <property name="toolTip">
           <string>Multiplier applied before time is sent as Fade Out Time to the function.</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Duration factor (+tap)</string>
          </property>
          <property name="toolTip">
           <string>Multiplier applied before time is sent as Duration to the function.</string>
          </property>
         </column>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="Input">
      <attribute name="title">
       <string>Input</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <widget class="QGroupBox" name="groupBox_2">
         <property name="title">
          <string>Absolute Value</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_4">
          <property name="topMargin">
           <number>4</number>
          </property>
          <property name="bottomMargin">
           <number>4</number>
          </property>
          <item row="0" column="0">
           <widget class="QLabel" name="m_absoluteRangeLabel">
            <property name="text">
             <string>Range</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QSpinBox" name="m_absoluteMinSpin"/>
          </item>
          <item row="0" column="2">
           <widget class="QSpinBox" name="m_absoluteMaxSpin"/>
          </item>
          <item row="0" column="3">
           <widget class="QCheckBox" name="m_absolutePrecisionCb">
            <property name="text">
             <string>ms precision</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0" colspan="3">
           <layout class="QHBoxLayout" name="m_absoluteInputLayout"/>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBox_3">
         <property name="title">
          <string>Tap</string>
         </property>
         <layout class="QHBoxLayout" name="m_tapInputLayout"/>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBox_4">
         <property name="title">
          <string>Apply</string>
         </property>
         <layout class="QHBoxLayout" name="m_applyInputLayout"/>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer_2">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="Appearance">
      <attribute name="title">
       <string>Appearance</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QCheckBox" name="m_pmCheck">
         <property name="text">
          <string>Show plus and minus buttons</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="m_dialCheck">
         <property name="text">
          <string>Show the central dial</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="m_tapCheck">
         <property name="text">
          <string>Show the tap button</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="m_hoursCheck">
         <property name="text">
          <string>Show the hours field</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="m_minCheck">
         <property name="text">
          <string>Show the minutes field</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="m_secCheck">
         <property name="text">
          <string>Show the seconds field</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="m_msCheck">
         <property name="text">
          <string>Show the milliseconds field</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="m_mdCheck">
         <property name="text">
          <string>Show multiplier and divisor buttons</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="m_applyCheck">
         <property name="text">
          <string>Show the apply button</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer_3">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>256</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="MultDiv">
      <attribute name="title">
       <string>Multiplier</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_4">
       <item>
        <widget class="QCheckBox" name="m_resetFactorOnDialChangeCb">
         <property name="text">
          <string>Reset multiplier factor when the dial value changes</string>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="m_multInputLayout"/>
       </item>
       <item>
        <layout class="QHBoxLayout" name="m_divInputLayout"/>
       </item>
       <item>
        <layout class="QHBoxLayout" name="m_multDivResetInputLayout"/>
       </item>
       <item>
        <spacer name="verticalSpacer_4">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>0</width>
           <height>0</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_2">
      <attribute name="title">
       <string>Presets</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <layout class="QGridLayout" name="gridLayout_10">
         <item row="2" column="2">
          <widget class="QLineEdit" name="m_presetNameEdit">
           <property name="toolTip">
            <string>Preset name</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QLabel" name="label_2">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>Preset Name</string>
           </property>
          </widget>
         </item>
         <item row="9" column="0" colspan="3">
          <layout class="QVBoxLayout" name="m_presetInputLayout"/>
         </item>
         <item row="0" column="1" colspan="2">
          <widget class="QPushButton" name="m_addPresetButton">
           <property name="text">
            <string> Add preset</string>
           </property>
           <property name="icon">
            <iconset resource="../qlcui.qrc">
             <normaloff>:/edit_add.png</normaloff>:/edit_add.png</iconset>
           </property>
          </widget>
         </item>
         <item row="3" column="1" colspan="2">
          <widget class="SpeedDial" name="m_speedDialWidget" native="true"/>
         </item>
         <item row="0" column="0" rowspan="9">
          <widget class="QTreeWidget" name="m_presetsTree">
           <column>
            <property name="text">
             <string>Name</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>Value</string>
            </property>
           </column>
          </widget>
         </item>
         <item row="1" column="1" colspan="2">
          <widget class="QPushButton" name="m_removePresetButton">
           <property name="text">
            <string> Remove</string>
           </property>
           <property name="icon">
            <iconset resource="../qlcui.qrc">
             <normaloff>:/edit_remove.png</normaloff>:/edit_remove.png</iconset>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item row="2" column="0" colspan="2">
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>SpeedDial</class>
   <extends>QWidget</extends>
   <header>speeddial.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../qlcui.qrc"/>
  <include location="../qlcui.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>VCSpeedDialProperties</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>266</x>
     <y>429</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>VCSpeedDialProperties</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>334</x>
     <y>429</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_addButton</sender>
   <signal>clicked()</signal>
   <receiver>VCSpeedDialProperties</receiver>
   <slot>slotAddClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>376</x>
     <y>121</y>
    </hint>
    <hint type="destinationlabel">
     <x>396</x>
     <y>106</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_removeButton</sender>
   <signal>clicked()</signal>
   <receiver>VCSpeedDialProperties</receiver>
   <slot>slotRemoveClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>376</x>
     <y>167</y>
    </hint>
    <hint type="destinationlabel">
     <x>396</x>
     <y>163</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_copyFactorsButton</sender>
   <signal>clicked()</signal>
   <receiver>VCSpeedDialProperties</receiver>
   <slot>slotCopyFactorsClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>376</x>
     <y>213</y>
    </hint>
    <hint type="destinationlabel">
     <x>396</x>
     <y>223</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_pasteFactorsButton</sender>
   <signal>clicked()</signal>
   <receiver>VCSpeedDialProperties</receiver>
   <slot>slotPasteFactorsClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>376</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>396</x>
     <y>283</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>slotAddClicked()</slot>
  <slot>slotRemoveClicked()</slot>
 </slots>
</ui>
